### PR TITLE
[chore] Fix unreleased bug that JSON integers should support string or int

### PR DESCRIPTION
**Pull request by bogdandrutu**
Bug added in https://github.com/open-telemetry/opentelemetry-collector/pull/11275

### CODE REVIEW

File: pdata/ptrace/json.go [hunk 1]
```diff
@@ -113,7 +113,7 @@ func (dest Span) unmarshalJsoniter(iter *jsoniter.Iterator) {
 				iter.ReportError("readSpan.parentSpanId", fmt.Sprintf("parse parent_span_id:%v", err))
 			}
 		case "flags":
-			dest.orig.Flags = iter.ReadUint32()
+			dest.orig.Flags = json.ReadUint32(iter)
```
Comment (songy23) [inline pdata/ptrace/json.go:116]: Can we add a test for this?

Comment (bogdandrutu) [reply pdata/ptrace/json.go:116]: We have unit-tests for the `json.ReadUint32`. Or for what?

Comment (bogdandrutu) [reply pdata/ptrace/json.go:116]: https://github.com/open-telemetry/opentelemetry-collector/blob/main/pdata/internal/json/number_test.go#L67

Comment (songy23) [reply pdata/ptrace/json.go:116]: OK that's good

**Pull request merged**